client/client_test.go

```diff
@@ -0,0 +1,45 @@
+package client_test
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	ethereumtest "github.com/direct-state-transfer/dst-go/ethereum/test"
+
+	"github.com/stretchr/testify/require"
+	_ "perun.network/go-perun/backend/ethereum/wallet/test"
+
+	"github.com/direct-state-transfer/dst-go/client"
+)
+
+func Test_Integ_InitializeClient(t *testing.T) {
+	t.Run("valid", func(t *testing.T) {
+		rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+		s, cleanup := ethereumtest.NewContractBackendSetup(t, rng, "../testdata")
+		defer cleanup()
+
+		_, err := client.InitClient(s.ContractBackend, *s.User, s.AdjAddr, s.AssetAddr)
+		require.NoError(t, err)
+	})
+	t.Run("invalid-contracts", func(t *testing.T) {
+		rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+		s, cleanup := ethereumtest.NewContractBackendSetup(t, rng, "../testdata")
```

Comment:
You can use the same setup in all three sub-tests since you only change it in the last one.

Reply:
Updated.